### DISCUSSION TITLE
How do we make a case insensitive query?

### DISCUSSION THREAD

**Discussion by simonowy** [apache/age#122] [Q&A] [answered]
Is there a way to do case insensitive search in AGE? Like in NeoJS or Agensgraph's, where you can do, 

Match (m:Person) WHERE m.Name = ~'(?i)search_me.*' RETURN m.Name;

**Comment by jrgemignani** [comment]
Neither Agensgraph or Neo4j support the command that you provided. Could you provide a working example in either of those graph databases showing your expectation?

**Neo4j -**
```
Query:
Match (m:Person) WHERE m.Name = ~'(?i)search_me.*' RETURN m.Name;
Error: org.neo4j.graphdb.QueryExecutionException: Invalid input '~': expected whitespace, comment or an expression (line 1, column 41 (offset: 40))
"Match (m:Person) WHERE m.Name = ~'(?i)search_me.*' RETURN m.Name;"
```

**Agensgraph -**
```
11.5=# Match (m:Person) WHERE m.Name = ~'(?i)search_me.*' RETURN m.Name;
ERROR:  operator is not unique: ~ unknown
LINE 1: Match (m:Person) WHERE m.Name = ~'(?i)search_me.*' RETURN m....
                                        ^
HINT:  Could not choose a best candidate operator. You might need to add explicit type casts.
```

**Reply by simonowy** [reply]
Sorry John  my typo mistake. It is MATCH (m:person) WHERE m.name =~'(?i)search_me.*' RETURN m.name;
That is without a space between = and ~, it works the Agensgraph v2.1.1. My staffer got that WHERE clause code fragment from a NeoJS forum.
We are using it on Agensgraph v2.1.1 in our development and it works.

Attached below, please find a working example and it did show the results as expected:
![20210930_205744](https://user-images.githubusercontent.com/73688643/135460773-ac8239e2-fe77-465a-af7d-e1bc01bd7c75.jpg)

Thank you

**Comment by jrgemignani** [comment]
Thank you for the updated command and thank you for pointing this out.

Currently there isn't a way to do a case insensitive search like that.
However, as it is available in both Agensgraph and Neo4j, we will add it to
our list of tasks for AGE.

When we are able to work on it will depend on its complexity and relative
priority. If it is simple, we'll try to get it out by our next release.

John

**Reply by simonowy** [reply]
Thanks John. 

Most applications need case insensitive search nowadays. I believe, when this feature is in, it will definitely leap frog AGE on Postgresql as the choice for graph database for many of the applications.

Kind regards,

Simon

**Comment by jrgemignani** [comment, marked as answer]
Hi Simon.

I was able to add =~ into my development version of AGE. I still need to
get the edge cases and regression tests done. But, it looks like it will be
pushed to the master branch sometime by the middle of next week - provided
nothing goes wrong. So, anyone willing to build AGE will have access to it
shortly.

Otherwise, fate permitting, it should be broadly available in the next
release.

John

**Reply by simonowy** [reply]
Thank you John.

Fantastic! I will give that a go to build AGE end of next week.

Kind regards,

Simon

**Comment by jrgemignani** [comment]
This addition has been pushed to the master.

**Reply by simonowy** [reply]
Hi John,

I just got that compiled and tested, it works! 

Good works! Thank you.

Kind regards,

Simon
